biz.aQute.repository/src/aQute/bnd/repository/maven/pom/provider/BndPomRepository.java

```diff
@@ -166,49 +199,56 @@ public boolean refresh() throws Exception {
 
 	@Override
 	public void setProperties(Map<String, String> map) throws Exception {
-
-		configuration = Converter.cnv(PomConfiguration.class, map);
-
-		if (configuration.name() == null)
-			throw new IllegalArgumentException("Must get a name");
-
-		this.name = configuration.name();
-
-		if (configuration.pom() != null) {
-			pomFiles = Strings.split(configuration.pom())
-				.stream()
-				.map(part -> {
-					File f = IO.getFile(part);
-					return f.isFile() ? f.toURI() : URI.create(part);
-				})
-				.collect(toList());
-			if (pomFiles.isEmpty()) {
-				throw new IllegalArgumentException("Pom is neither a file nor a revision " + configuration.pom());
-			}
-		} else if (configuration.revision() != null) {
-			revisions = Strings.split(configuration.revision())
-				.stream()
-				.map(Revision::valueOf)
-				.filter(Objects::nonNull)
-				.collect(toList());
-			if (revisions.isEmpty()) {
-				throw new IllegalArgumentException(
-					"Revision is neither a file nor a revision " + configuration.revision());
-			}
-		} else if (configuration.query() != null) {
-			this.query = configuration.query();
-			this.queryUrl = configuration.queryUrl("http://search.maven.org/solrsearch/select");
-		} else {
-			throw new IllegalArgumentException("Neither pom, revision nor query property are set");
-		}
 		synchronized (this) {
+			configuration = Converter.cnv(PomConfiguration.class, map);
+
+			if (configuration.name() == null)
+				throw new IllegalArgumentException("Must get a name");
+
+			this.name = configuration.name();
+
+			if (configuration.pom() != null) {
+				pomFiles = Strings.split(configuration.pom())
+					.stream()
+					.map(part -> {
+						File f = IO.getFile(part);
+						return f.isFile() ? f.toURI() : URI.create(part);
+					})
+					.collect(toList());
+				if (pomFiles.isEmpty()) {
+					status("Pom is neither a file nor a revision " + configuration.pom());
+				}
+			} else if (configuration.revision() != null) {
+				revisions = Strings.split(configuration.revision())
+					.stream()
+					.map(Revision::valueOf)
+					.filter(Objects::nonNull)
+					.collect(toList());
+				if (revisions.isEmpty()) {
+					status("Revision is neither a file nor a revision " + configuration.revision());
+				}
+			} else if (configuration.query() != null) {
+				this.query = configuration.query();
+				this.queryUrl = configuration.queryUrl("http://search.maven.org/solrsearch/select");
+			} else {
+				status("Neither pom, revision nor query property are set");
+			}
 			inited = false;
```

Comment:
don't need to do this. setProperties is only called once

Reply:
you mean the `inited = false` here?